### DISCUSSION TITLE
6.2.0 (Grid effects - Experimental)

### DISCUSSION THREAD

**Discussion by T-Troll** [T-Troll/alienfx-tools#152] [Announcements]
### Changes in this release:

`AlienFX-GUI`:
- **EXPERIMENTAL:** Grid effect mode added as well as corresponding tab. It's like "Global effect", but software-controlled. Currently, into Alpha stage, so now it's just for the enthusiasts who want to take a look and share opinion. Please read [this wiki page](https://github.com/T-Troll/alienfx-tools/wiki/Grid-Effects) about how it works.
- Interface tuning and small enchantments, especially at Ambient tab (for better reading of grid buttons).
- Bug fixed: Incorrect gauge zone calculation in some cases.
- Bug fixed: Speed/Length sliders doesn't work correctly at "Colors" tab

All fan-control related apps and `AlienFX-Mon`:
- New temperature sensor source added into detection configuration - ECDV sensors. Available on modern G-series systems and some Alienware.

`AlienFan-GUI`:
- New: Tool tip added to application tray icon, displaying current power mode and fan RPMs.
- Temperature/RPM/Curve update optimized, decreasing CPU load then application not minimized. The same for `AlienFX-GUI` with active "Fans and Power" tab.
- Bug fixed: A bunch of bugs related to initial overboost calculation.
- Bug fixed: right click on icon does not close application.

Known issues and limitations:
- Grid effects settings not saved or loaded between session (data structures not final yet, and will be changed).
- Some grid effect modes and triggers not implemented yet.
- ECDV sensor names very hard to retrieve (random device names), so it's just numbered for now. I'm working on it.

<hr /><em>This discussion was created from the release <a href='https://github.com/T-Troll/alienfx-tools/releases/tag/6.2.0'>6.2.0 (Grid effects - Experimental)</a>.</em>

**Comment by MarlosLucas** [comment]
I really do not understand why suddenly stop working in M15 R6 disabling the fan control... AWCC is fully unistalled but very commonly, I face this issue.

**Reply by T-Troll** [reply]
This happened at some W11 system, seems like after system update.
Please read this - https://github.com/T-Troll/alienfx-tools/issues/114

**Reply by MarlosLucas** [reply]
Thanks, T-Troll. Good job, by the way.

Let me clearer for your better understanding. A few hours ago, It was working fine. But, after some simulations (data science bla bla bla), Ive faced a BSOD. After the windows start, the activation of fan control is impossible ("Fan control start failure, disabling fan control!").

**Reply by T-Troll** [reply]
Thank you....

Ok, got it. By the way, do you remember BSOD reason? I need to check is it my driver-related or other system component.
This error message means it can't load low-level driver.
So first check loader and driver still in place and don't remove by your AV (check app folder, kdl.dll and hwacc.sys can present inside). Also check AV settings - app folder should be in exception list.
Second, please check is any Windows updates installed last time (then it stops working) - this help us to pinpoint if issue into update.
Third, check your security settings - "Memory integrity" should be disabled.
And, of course, just restart and try again, in case you don't do so (but better check p.1 first).

PS: Maybe it's reasonable to open an issue about it.

**Reply by MarlosLucas** [reply]
Hey, T-Troll! Im sorry about the absense, man.

Every step that you mentioned its ok here. What I did at last was to install the last version and it works fine.

Unfortunately, the issue returned. Ive tried to return the lastest version and even reinstall the other version but the fan control is still unavailable.

My main difficult is to find a pattern to happen this. I still believe that is related to windows defender and all this stull. I really dont know...

Cheers!!

**Reply by T-Troll** [reply]
Yes, it's looks like Defender settings.
Did the files mention above (kdl.dll and hwacc.sys) still into application folder and not blocked by AV?
Did app folder into AV exceptions list?
You can try to run gui from explorer window (not menu shortcut) and watch - 2 other files should be created at fan start attempt (and then deleted).
Maybe i build and share a special debug app for these issues, W11 is a pain for many ppl....

**Reply by MarlosLucas** [reply]
Yep
Yep
Yep
Same...
Please!!!!

Have another tip? I can do it now

**Reply by T-Troll** [reply]
You can try to download [KDU](https://github.com/hfiref0x/KDU) and play with it (different providers), as described in issue discussion i share with you above.
My loader based on it.

**Comment by timdawg** [comment]
Windows Defender is detecting this release as Trojan:Win32/Woreflint.A!cl

I tried both the MSI and ZIP files

**Reply by T-Troll** [reply]
It's not a trojan, i mention this into readme (maybe need bigger font...).

Being short, MS is not friendly to open-source developers. From W10 release 1907, they require sign for every driver, and it's hard (and not cheap) to obtain.

So i use hack-way (BTW, well-known from XP times) to load low-level driver for fan control.

Antivirus complain about driver loader dll as a result. It's not exactly false-positive, but the case AV decide this is "unwanted" for user.

Both loader and driver source code available here, so you can check what it does and build yourself to be sure.

**Comment by SamSamCk** [comment]
hi, i have some problem with my alienware m15R2 laptop. 
1- alienfx-gui is not working at all. None of colors are changing and the back alien head + tron light is off and i can't reactivate.
2 G-mode is not usable
3 if i remove the program and reinstall it its the same problem because the configuration is still here.
i like this application and the effort put in this project, but i have some problem... can someone help me please?

**Reply by T-Troll** [reply]
Can you explain, please?
1. Can you MANUALLY assign colors to the grid (i don't have R2 data in database)?
2. G-Mode doesn't supported at most of the Alienware. Even if supported in BIOS, it can do nothing (like GPU mode in some of them). Use one of the Power modes instead - this is analogue of "Thermal" into AWCC.
3. Remove: `HKCU/Software/Alienfx_sdk`, `HKCU/Software/Alienfan`, `HKCU/Software/aliefxgui` from registry. This is all configuration changes it has.

Calm down, better tell me what you expect and what you have, and i try to help.

**Reply by SamSamCk** [reply]
oh ok i've forgot to remove the regedit key.
sorry, i diddn't want to look upset.
1. i don't know how to do that. Can you explain me how to do it ? (i can post some screenshot if need it)
2. its ok its just a detail
3. i've done that but its the same thing. (not important)
sorry for the late response

**Reply by T-Troll** [reply]
1. Did you read release notes? It's a link to [beginner's guide](https://github.com/T-Troll/alienfx-tools/wiki/How-to-start-%28Beginner%27s-guide%29-for-release-v6.x.x.x) there. Read this document, please!
